deltatech_product_tracking/models/product.py

```diff
@@ -17,61 +30,63 @@ def onchange_tracking(self):
     def write(self, vals):
         res = super(ProductProduct, self).write(vals)
         if vals.get("tracking", "none") == "lot":
-            warehouses = self.env["stock.warehouse"].with_context(active_test=False).search([])
-            for product in self:
-                # Parse warehouses
-                company = product.company_id or self.env.user.company_id
-                for war in warehouses:
-                    war_loc = war.lot_stock_id
-                    domain = [
-                        ("product_id", "=", product.id),
-                        "|",
-                        ("location_id", "=", war_loc.id),
-                        ("location_dest_id", "=", war_loc.id),
-                    ]
-                    st_moves = self.env["stock.move"].search(domain)
+            self.fill_lot_in_stock_docs()
+        return res
 
-                    if st_moves:
-                        lots = st_moves.mapped("lot_ids")
-                        if lots:
-                            lot = lots[0]
-                        else:
-                            lot_name = war.code + "000001"
-                            lot = self.env["stock.production.lot"].search(
-                                [
-                                    ("name", "=", lot_name),
-                                    ("product_id", "=", product.id),
-                                    ("company_id", "=", company.id),
-                                ],
-                                limit=1,
-                            )
-                            if not lot:
-                                lot = self.env["stock.production.lot"].create(
-                                    {
-                                        "name": lot_name,
-                                        "product_id": product.id,
-                                        "company_id": company.id,
-                                    }
-                                )
-                        if lot:
-                            domain = [("lot_id", "=", False)] + domain
-                            st_move_lines = self.env["stock.move.line"].search(domain)
+    def fill_lot_in_stock_docs(self):
+        warehouses = self.env["stock.warehouse"].with_context(active_test=False).search([])
```

Comment:
gasim o varinate pe history, sa actualizam si quant-uri di nalte locatii, adica daca avem transfer prin transit, si quant-urile sunt in tranzit in momentul asta...